shop/src/main/java/ro/msg/learning/shop/Services/OrderServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package ro.msg.learning.shop.Services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTOs.orderDto.OrderDTOInput;
+import ro.msg.learning.shop.DTOs.orderDto.OrderDTOOutput;
+import ro.msg.learning.shop.Repositories.OrderRepository;
+import ro.msg.learning.shop.configuration.OrderStrategyConfiguration;
+import ro.msg.learning.shop.exceptions.OrderPlacingException;
+import ro.msg.learning.shop.mappers.OrderMapper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements IOrderService {
```

Comment:
use @Transactional on services with multiple JPA calls if you want them all to happen in a single DB transaction. So, for example, saving a result in two tables, and you don’t want it saving one if the other fails, in which case both will be rolled back on failure.